gui/src/components/JobForm.js

```diff
@@ -0,0 +1,95 @@
+import React, { Fragment } from 'react'
+import { withFormik, Form } from 'formik'
+import { withStyles } from '@material-ui/core/styles'
+import Button from '@material-ui/core/Button'
+import { TextField, Grid } from '@material-ui/core'
+import { connect } from 'react-redux'
+import { submitJobSpec } from 'actions'
+import matchRouteAndMapDispatchToProps from 'utils/matchRouteAndMapDispatchToProps'
+import { Prompt } from 'react-static'
+import { BridgeAndJobNotifications } from './FormNotifications'
+
+const styles = theme => ({
+  jsonfield: {
+    paddingTop: theme.spacing.unit * 1.25,
+    width: theme.spacing.unit * 150
```

Comment:
This should use the grid instead of trying to set a width. On smaller screens this causes the text field to overflow.

<img width="947" alt="screen shot 2018-08-20 at 9 50 28 am" src="https://user-images.githubusercontent.com/680789/44354881-df4cc980-a45f-11e8-86e0-983a406b8063.png">

Reply:
@rupurt good catch. fixed this and wondering if the same should be done on BridgeForm. (bridgeform inputs have small width by default and I doubt this can ever be a problem there)

Reply:
@NavyAdmiral yeah, I think we should also do it for bridge types. Though I think we should try to make the text fields roughly the same size they are now, even on bigger screens. You can do that by using the `xs` & `xl` props on the `<Grid item>...` component.

I also feel like we should make the create job textarea wider.